lib/sitemap_generator/link_set.rb

```diff
@@ -367,6 +367,10 @@ def yield_sitemap?
       @yield_sitemap.nil? ? SitemapGenerator.yield_sitemap? : !!@yield_sitemap
     end
 
+    def max_sitemap_links
```

Comment:
Rather just use an attr_accessor and assign the default value in `initialize` method like we do for the options.  See LinkSet line 118-129.

Then in `sitemap` method line 312-314 pass the `@max_sitemap_links` attribute to the `SitemapGenerator::Builder::SitemapFile` initializer and use that option within SitemapFile.

Also add `:max_sitemap_links` to `current_settings` array in `options_for_group` method so that groups inherit the option.